lib-injection/sources/sitecustomize.py

```diff
@@ -304,34 +360,58 @@ def _inject():
                 )
 
         # check installed packages against allow list
-        incompatible_packages = {}
-        for package_name, package_version in INSTALLED_PACKAGES.items():
-            if not package_is_compatible(package_name, package_version):
-                incompatible_packages[package_name] = package_version
+        incompatible_dependencies = {} # Store incompatible core dependencies
+        disabled_integrations = {} # Store {integration_name: (pkg_name, installed_version)}
 
-        if incompatible_packages:
-            _log("Found incompatible packages: %s." % incompatible_packages, level="debug")
-            integration_incomp = True
-            if not FORCE_INJECT:
-                _log("Aborting dd-trace-py instrumentation.", level="debug")
-                abort = True
-
-                for key, value in incompatible_packages.items():
+        for package_name, package_version in INSTALLED_PACKAGES.items():
+            # Check if it's an integration package
+            integration_info = MIN_INTEGRATION_PKGS.get(package_name)
+            if integration_info:
+                if not integration_package_is_compatible(package_version, integration_info):
+                    integration_name = integration_info.integration_name
+                    _log(f"Found incompatible integration package: {package_name} v{package_version} (requires >= {integration_info.min_version.version}, constraint: '{integration_info.min_version.constraint}'). Disabling integration '{integration_name}'.", level="debug")
```

Comment:
Don't we end up logging here and on line 407, both saying that we found incompatible integrations and are disabling them? Shouldn't we only say we're disabling them on 407, because here we actually don't know if we're going to disable it or not (since we're not checking FORCE_INJECT)